RELEASE/scripts/autoscend/auto_restore.ash

```diff
@@ -1674,6 +1674,64 @@ boolean __restore(string resource_type, int goal, int meat_reserve, boolean useF
 	return true;
 }
 
+void auto_beaten_handler()
+{
+	void __remove_beaten_up()
+	{
+		if(get_property("auto_beatenUpCount").to_int() <= 10 && my_mp() >= mp_cost($skill[Tongue of the Walrus]) && auto_have_skill($skill[Tongue of the Walrus]))
+		{
+			auto_log_info("trying to recover with [Tongue of the Walrus]", "red");
+			use_skill(1, $skill[Tongue of the Walrus]);
+			if(have_effect($effect[Beaten Up]) == 0)
+			{
+				return;
+			}
+			else
+			{
+				auto_log_warning("Mysteriously failed to recover beaten up with [Tongue of the Walrus]");
+			}
+		}
+	}
+
```

Comment:
I moved the removal code itself here so that it's still done before the early return from the June Cleaver case where we don't want to increment auto_beatenUpCount.